tests/unit/schema/test_api.py

```diff
@@ -2602,3 +2604,215 @@ def test_has_iam_mixin():
     })
     api_schema = api.API.build(fd, 'google.example.v1', opts=opts)
     assert api_schema.has_iam_mixin
+
+
+def test_all_method_settings():
+    """
+    Tests the `all_method_settings` method of `gapic.schema.api` which
+    reads `MethodSettings` from the service config YAML.
+    https://github.com/googleapis/googleapis/blob/7dab3de7ec79098bb367b6b2ac3815512a49dd56/google/api/client.proto#L325
+    """
+    field_options = descriptor_pb2.FieldOptions()
+    field_options.Extensions[
+        field_info_pb2.field_info
+    ].format = field_info_pb2.FieldInfo.Format.Value("UUID4")
+
+    # See AIP https://google.aip.dev/client-libraries/4235
+    # for the specific requirements to be tested.
+    squid = make_field_pb2(
+        name="squid", type="TYPE_STRING", options=field_options, number=1
+    )
+    mollusc = make_field_pb2(
+        name="mollusc", type="TYPE_STRING", options=field_options, number=2
+    )
+    # Also include a required field
+    field_options.Extensions[field_behavior_pb2.field_behavior].append(
+        field_behavior_pb2.FieldBehavior.Value("REQUIRED")
+    )
+    clam = make_field_pb2(
+        name="clam", type="TYPE_STRING", options=field_options, number=3
+    )
+    # Also include a field which is not of type string
```

Comment:
This tests both non-string and a non-UUID4. I suggest separate test cases.

Reply:
Fixed in 5a2d33b8b160a246945e3752e145878581a351b1